packages/react-aria-components/src/GridList.tsx

```diff
@@ -561,3 +563,62 @@ export const GridListLoadMoreItem = createLeafComponent('loader', function GridL
     </>
   );
 });
+
+export interface GridListSectionProps<T> extends SectionProps<T> {}
+
+function GridListSectionInner<T extends object>(props: GridListSectionProps<T>, ref: ForwardedRef<HTMLElement>, section: Node<T>, className = 'react-aria-GridListSection') {
+  let state = useContext(ListStateContext)!;
+  let {dragAndDropHooks, dropState} = useContext(DragAndDropContext)!;
+  let {CollectionBranch} = useContext(CollectionRendererContext);
+  let [headingRef, heading] = useSlot();
+  let {headingProps, rowProps, rowGroupProps} = useGridListSection({
+    heading,
+    'aria-label': props['aria-label'] ?? undefined
+  });
+  let renderProps = useRenderProps({
+    defaultClassName: className,
+    className: props.className,
+    style: props.style,
+    values: {}
+  });
+
+  return (
+    <section
+      {...filterDOMProps(props as any)}
+      {...rowGroupProps}
+      {...renderProps}
+      ref={ref}>
+      <Provider
+        values={[
+          [HeaderContext, {...headingProps, ref: headingRef}],
+          [GridListHeaderContext, {...rowProps}]
+        ]}>
+        <CollectionBranch
+          collection={state.collection}
+          parent={section}
+          renderDropIndicator={useRenderDropIndicator(dragAndDropHooks, dropState)} />
+      </Provider>
+    </section>
+  );
+}
+
+const GridListHeaderContext = createContext<HTMLAttributes<HTMLElement> | null>(null);
+
+export const GridListHeader = /*#__PURE__*/ createLeafComponent('header', function Header(props: HTMLAttributes<HTMLElement>, ref: ForwardedRef<HTMLElement>) {
```

Comment:
do we need a special one for GridList? I assume it has to do with the extra row div? would it make more sense as two components instead?

It's a little weird because ordinarily the ref would attach to the outer most element. This also makes it hard to style the div since there's no classname on it